### PR TITLE
ADM-718[frontend][stub]feat: Display verification results of ‘board’

### PR DESCRIPTION
**Pull request by K-Chow**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 1]
```diff
@@ -8,36 +8,51 @@ import {
   RESET,
   VERIFIED,
   VERIFY,
-  VERIFY_ERROR_MESSAGE,
-  VERIFY_FAILED,
 } from '../../fixtures';
-import { fireEvent, render, screen, waitFor, within } from '@testing-library/react';
+import {
+  fireEvent,
+  getByLabelText,
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:14]: 'getByLabelText' is defined but never used
BTW, 'NO_CARD_ERROR_MESSAGE' is defined but never used

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 2]
```diff
@@ -8,36 +8,51 @@ import {
...
+  render,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:18]: 'waitForElementToBeRemoved' is defined but never used

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 3]
```diff
@@ -8,36 +8,51 @@ import {
...
-  });
-  fieldInputs.map((input, index) => {
-    expect(input.value).toEqual(mockInfo[index]);
-  });
+
+export const fillBoardFieldsInformation = async () => {
+  await userEvent.type(screen.getByLabelText(/board id/i), '1');
+  await userEvent.type(screen.getByLabelText(/email/i), 'fake@qq.com');
+  await userEvent.type(screen.getByLabelText(/site/i), 'fake');
+  await userEvent.type(screen.getByLabelText(/token/i), 'fake-token');
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:33]: define a FAKE_TOKEN in the fixture, and refactoe existing code thx

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 4]
```diff
@@ -64,18 +79,17 @@ describe('Board', () => {
   });
 
   it('should show default value jira when init board component', () => {
-    setup();
-    const boardType = screen.getByText(BOARD_TYPES.JIRA);
+    const { getByRole } = setup();
+    const boardType = getByRole('button', {
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:83]: Avoid destructuring queries from `render` result, use `screen.getByRole` instead

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 5]
```diff
@@ -84,13 +98,22 @@ describe('Board', () => {
   });
 
   it('should show board type when select board field value ', async () => {
-    setup();
+    const { getByRole, getByText } = setup();
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:101]: Avoid destructuring queries from `render` result, use `screen.getByRole` instead

Comment (guzhongren) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx:101]: 'getByText' is assigned a value but never used

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 6]
```diff
@@ -84,13 +98,22 @@ describe('Board', () => {
...
 
-    fireEvent.mouseDown(screen.getByRole('button', { name: CONFIG_TITLE.BOARD }));
-    fireEvent.click(screen.getByText(BOARD_TYPES.CLASSIC_JIRA));
+    await userEvent.click(getByRole('button', { name: /board jira/i }));
 
     await waitFor(() => {
-      expect(screen.getByText(BOARD_TYPES.CLASSIC_JIRA)).toBeInTheDocument();
+      expect(getByRole('option', { name: /jira/i })).toBeInTheDocument();
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:106]: Avoid destructuring queries from `render` result, use `screen.getByRole` instead

See the detail in PR diff parts

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 7]
```diff
@@ -190,44 +219,23 @@ describe('Board', () => {
...
-        res(ctx.status(HttpStatusCode.Unauthorized), ctx.json({ hintInfo: VERIFY_ERROR_MESSAGE.UNAUTHORIZED })),
-      ),
-    );
+    server.use(rest.post(MOCK_BOARD_URL_FOR_JIRA, (_, res, ctx) => res(ctx.status(HttpStatusCode.Unauthorized))));
     setup();
-    fillBoardFieldsInformation();
+    await fillBoardFieldsInformation();
 
-    fireEvent.click(screen.getByRole('button', { name: VERIFY }));
+    fireEvent.click(screen.getByRole('button', { name: /verify/i }));
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:235]: avoid to use `fireEvent`, please using `userEvent`

File: frontend/__tests__/containers/ConfigStep/ConfigStep.test.tsx [hunk 8]
```diff
@@ -139,8 +139,8 @@ describe('ConfigStep', () => {
 
     fireEvent.mouseDown(screen.getByRole('button', { name: REQUIRED_DATA }));
     const requireDateSelection = within(screen.getByRole('listbox'));
-    fireEvent.click(requireDateSelection.getByRole('option', { name: VELOCITY }));
-    fireEvent.click(requireDateSelection.getByRole('option', { name: CYCLE_TIME }));
+    fireEvent.click(requireDateSelection.getByRole('option', { name: /velocity/i }));
+    fireEvent.click(requireDateSelection.getByRole('option', { name: /cycle time/i }));
```
Comment (guzhongren) [inline frontend/__tests__/containers/ConfigStep/ConfigStep.test.tsx:143]: userEvent

File: frontend/__tests__/containers/MetricsStep/MetricsStep.test.tsx [hunk 9]
```diff
@@ -248,5 +250,22 @@ describe('MetricsStep', () => {
...
     });
+
+    it('when get board card when no data then should be render no card container', async () => {
+      server.use(
+        rest.post(MOCK_BOARD_INFO_URL, (_, res, ctx) => {
+          return res(ctx.status(HttpStatusCode.Ok));
+        }),
+      );
+
+      const { getByText } = setup();
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStep/MetricsStep.test.tsx:261]: so many the same issues, please update

File: frontend/__tests__/containers/MetricsStepper/MetricsStepper.test.tsx [hunk 10]
```diff
@@ -309,7 +312,7 @@ describe('MetricsStepper', () => {
   it('should export json when click save button when pipelineTool, sourceControl, and board is not empty', async () => {
     const expectedFileName = 'config';
     const expectedJson = {
-      board: { boardId: '', email: '', projectKey: '', site: '', token: '', type: 'Jira' },
+      board: { boardId: '', email: '', projectKey: '', site: '', token: '', type: 'Jira', startTime: 0, endTime: 0 },
```
Comment (guzhongren) [inline frontend/__tests__/containers/MetricsStepper/MetricsStepper.test.tsx:315]: duplicated code

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 11]
```diff
@@ -1,44 +1,150 @@
...
+import { rest } from 'msw';
+
+const mockDispatch = jest.fn();
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useDispatch: () => mockDispatch,
+}));
+
+jest.mock('@src/hooks/useAppDispatch', () => ({
+  useAppSelector: () => ({ type: 'Jira' }),
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:17]: use the `BOARD_TYPES.JIRA`, not customized

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 12]
```diff
@@ -1,44 +1,150 @@
...
+        type: 'jira',
+        boardId: '1',
+        site: 'fake',
+        token: 'fake-token',
+        startTime: null,
+        endTime: null,
+      });
     });
+
+    await waitFor(() => {
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:78]: Avoid using multiple assertions within `waitFor` callback

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 13]
```diff
@@ -1,44 +1,150 @@
...
+const server = setupServer();
+
 describe('use verify board state', () => {
-  it('should initial data state when render hook', async () => {
+  beforeAll(() => server.listen());
+  afterAll(() => {
+    jest.clearAllMocks();
+    server.close();
+  });
+  it('when hook render given none input  then should got initial data state ', async () => {
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:28]: update your test description with our UT convension `should .... when ...given`

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 14]
```diff
@@ -1,44 +1,150 @@
...
+        endTime: null,
+      });
+    });
+
+    await waitFor(() => {
+      const site = result.current.formFields.find((field) => field.name === 'site');
 
-    act(() => {
-      result.current.verifyJira(MOCK_BOARD_VERIFY_REQUEST_PARAMS);
+      expect(site?.errorMessage).toBe('Site is incorrect !');
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:116]: update the whole test description

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 15]
```diff
@@ -1,44 +1,150 @@
...
+    await act(() => {
+      result.current.verifyJira({
+        type: 'jira',
+        boardId: '1',
+        site: 'fake',
+        token: 'fake-token',
+        startTime: null,
+        endTime: null,
+      });
+    });
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:111]: duplicated code

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 16]
```diff
@@ -1,44 +1,150 @@
...
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:142]: duplicated code

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx [hunk 17]
```diff
@@ -1,44 +1,150 @@
...
+
+    await waitFor(() => {
+      const boardId = result.current.formFields.find((field) => field.name === 'boardId');
+
+      expect(boardId?.errorMessage).toBe('Board Id is incorrect !');
```
Comment (guzhongren) [inline frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:147]: are u sure there is a blank between `!` and `incorrect`?

File: frontend/cypress/e2e/createANewProject.cy.ts [hunk 18]
```diff
@@ -249,6 +249,8 @@ describe('Create a new project', () => {
     });
   });
 
+  // just ignore for bug bash , need ti fix next time @Zhou Kai, @Xingmeng Tao
```
Comment (guzhongren) [inline frontend/cypress/e2e/createANewProject.cy.ts:252]: ok, todo

File: frontend/cypress/e2e/createANewProject.cy.ts [hunk 19]
```diff
@@ -318,6 +320,7 @@ describe('Create a new project', () => {
 
     configPage.nextStepButton.should('be.enabled');
 
+    /*
```
Comment (guzhongren) [inline frontend/cypress/e2e/createANewProject.cy.ts:323]: please give todo who responsible for this change, if you know

File: frontend/src/components/Common/EmptyContent/index.tsx [hunk 20]
```diff
@@ -0,0 +1,21 @@
+/* istanbul ignore file */
```
Comment (guzhongren) [inline frontend/src/components/Common/EmptyContent/index.tsx:1]: don't ignore the whole file, if it is not for bug bash

File: frontend/src/constants/resources.ts [hunk 21]
```diff
@@ -73,7 +73,6 @@ export enum CONFIG_TITLE {
 }
 
 export const BOARD_TYPES = {
-  CLASSIC_JIRA: 'Classic Jira',
```
Comment (guzhongren) [inline frontend/src/constants/resources.ts:76]: are u sure?  we have plan our HB will not support `Classic Jira`, just yes or no?

File: frontend/src/constants/resources.ts [hunk 22]
```diff
@@ -178,6 +177,10 @@ export enum REPORT_SUFFIX_UNITS {
 
 export const MESSAGE = {
   VERIFY_FAILED_ERROR: 'verify failed',
+  VERIFY_MAIL_FAILED_ERROR: 'Email is incorrect !',
```
Comment (guzhongren) [inline frontend/src/constants/resources.ts:180]: There is a blank between `!` and `incorrect`? 
sam issue as below

File: frontend/src/containers/MetricsStep/index.tsx [hunk 23]
```diff
@@ -29,9 +47,21 @@ const MetricsStep = ({ closeAllNotifications }: useNotificationLayoutEffectInter
     requiredData.includes(REQUIRED_DATA.CLASSIFICATION);
   const isShowRealDone =
     cycleTimeSettingsType === CYCLE_TIME_SETTINGS_TYPES.BY_COLUMN && cycleTimeSettings.some((e) => e.value === DONE);
+  const { getBoardInfo, isLoading, errorMessage } = useGetBoardInfoEffect();
+
+  /* istanbul ignore next */
```
Comment (guzhongren) [inline frontend/src/containers/MetricsStep/index.tsx:52]: todo

File: frontend/src/context/config/configSlice.ts [hunk 24]
```diff
@@ -97,7 +97,8 @@ export const configSlice = createSlice({
             ? null
             : MESSAGE.CONFIG_PAGE_VERIFY_IMPORT_ERROR;
       }
-      state.board.config = action.payload.board || state.board.config;
+      /* istanbul ignore next */
```
Comment (guzhongren) [inline frontend/src/context/config/configSlice.ts:100]: todo

File: frontend/src/hooks/useGetBoardInfo.ts [hunk 25]
```diff
@@ -0,0 +1,71 @@
+/* istanbul ignore file */
```
Comment (guzhongren) [inline frontend/src/hooks/useGetBoardInfo.ts:1]: todo

File: frontend/src/hooks/useVerifyBoardEffect.ts [hunk 26]
```diff
@@ -1,44 +1,178 @@
...
+        const { message, code } = e;
+        if (code === 401) {
+          setErrorField(['email', 'token'], [MESSAGE.VERIFY_MAIL_FAILED_ERROR, MESSAGE.VERIFY_TOKEN_FAILED_ERROR]);
+        }
+        if (code === 404 && message === 'site not found') {
+          setErrorField(['site'], [MESSAGE.VERIFY_SITE_FAILED_ERROR]);
+        }
+        if (code === 404 && message === 'boardId not found') {
+          setErrorField(['boardId'], [MESSAGE.VERIFY_BOARD_FAILED_ERROR]);
+        }
```
Comment (guzhongren) [inline frontend/src/hooks/useVerifyBoardEffect.ts:165]: 401 -> use httpcode enum

File: stubs/backend/jira/jira-stubs.yaml [hunk 27]
```diff
@@ -77,26 +126,33 @@
...
+    file: ./backend/jira/jsons/jira.board.info.configuration.json
+# - request:
+#     method: GET
+#     url: /rest/api/2/status/(PLL-\d+)
+
+#   response:
+#     headers:
+#       content-type: application/json
+#     status: 200
+#     file: ./backend/jira/jsons/jira.board.info.status.json
```
Comment (guzhongren) [inline stubs/backend/jira/jira-stubs.yaml:158]: just delete it if we don't use it

**Review by guzhongren — COMMENTED** [review]
change with the comments

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx (hunk 1 above)
Comment (K-Chow) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx]: fixed

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx (hunk 3 above)
Comment (K-Chow) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx]: fixed

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx (hunk 4 above)
Comment (K-Chow) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx:75]: fixed

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx (hunk 13 above)
Comment (K-Chow) [reply frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx]: fixed

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx (hunk 14 above)
Comment (K-Chow) [reply frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx]: fixed

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx (hunk 15 above)
Comment (K-Chow) [reply frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx]: fixed

File: frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx (hunk 17 above)
Comment (K-Chow) [reply frontend/__tests__/hooks/useVerifyBoardEffect.test.tsx:126]: yes, i'm sure

File: frontend/src/components/Common/EmptyContent/index.tsx (hunk 20 above)
Comment (K-Chow) [reply frontend/src/components/Common/EmptyContent/index.tsx:2]: will coverage next time , and already assigned to me

File: frontend/src/containers/MetricsStep/index.tsx (hunk 23 above)
Comment (K-Chow) [reply frontend/src/containers/MetricsStep/index.tsx]: fixed

File: frontend/src/hooks/useVerifyBoardEffect.ts (hunk 26 above)
Comment (K-Chow) [reply frontend/src/hooks/useVerifyBoardEffect.ts]: fixed

File: stubs/backend/jira/jira-stubs.yaml (hunk 27 above)
Comment (K-Chow) [reply stubs/backend/jira/jira-stubs.yaml]: ok, removed

File: frontend/src/hooks/useGetBoardInfo.ts (hunk 25 above)
Comment (K-Chow) [reply frontend/src/hooks/useGetBoardInfo.ts]: fixed

File: frontend/src/constants/resources.ts (hunk 21 above)
Comment (SimonTal) [reply frontend/src/constants/resources.ts:76]: We still support `Classic Jira`. It's just that there is no need for the user to select it manually, but the backend acquires the type directly through API.

File: frontend/src/containers/MetricsStep/Crews/index.tsx [hunk 28]
```diff
@@ -20,9 +20,13 @@ export const Crews = ({ options, title, label, type = 'board' }: crewsProps) =>
   const dispatch = useAppDispatch();
   const [isEmptyCrewData, setIsEmptyCrewData] = useState<boolean>(false);
   const { users, pipelineCrews } = useAppSelector(selectMetricsContent);
-  const [selectedCrews, setSelectedCrews] = useState(isBoardCrews ? users : pipelineCrews);
+  const [selectedCrews, setSelectedCrews] = useState<string[]>([]);
   const isAllSelected = options.length > 0 && selectedCrews.length === options.length;
 
+  useEffect(() => {
```
Comment (SimonTal) [inline frontend/src/containers/MetricsStep/Crews/index.tsx:26]: It seems that you can replace this `useEffect/setState` pattern with a useMemo.

File: frontend/src/hooks/useGetBoardInfo.ts [hunk 29]
```diff
@@ -0,0 +1,74 @@
...
+    setErrorMessage({});
+    return boardInfoClient
+      .getBoardInfo({
+        ...data,
+        token: getJiraBoardToken(data.token, data.email),
+      })
+      .then((res) => {
+        if (!res.data) {
+          setErrorMessage({
+            title: 'No card within selected date range!',
```
Comment (SimonTal) [inline frontend/src/hooks/useGetBoardInfo.ts:56]: extract wordings into resource file

File: frontend/src/hooks/useVerifyBoardEffect.ts [hunk 30]
```diff
@@ -1,44 +1,183 @@
...
+      .then((result) => {
+        clearError();
+        return result;
+      })
+      .catch((e) => {
+        const { message, code } = e;
+        if (code === HttpStatusCode.Unauthorized) {
+          setErrorField(['email', 'token'], [MESSAGE.VERIFY_MAIL_FAILED_ERROR, MESSAGE.VERIFY_TOKEN_FAILED_ERROR]);
+        }
+        if (code === HttpStatusCode.NotFound && message === 'site not found') {
```
Comment (SimonTal) [inline frontend/src/hooks/useVerifyBoardEffect.ts:165]: extract message as const or object mapping

File: frontend/src/hooks/useGetBoardInfo.ts (hunk 29 above)
Comment (K-Chow) [reply frontend/src/hooks/useGetBoardInfo.ts]: fixed

File: frontend/__tests__/components/Common/EmptyContent/EmptyContent.test.tsx [hunk 31]
```diff
@@ -0,0 +1,10 @@
+import EmptyContent from '@src/components/Common/EmptyContent';
+import { render, screen } from '@testing-library/react';
+
+describe('EmptyContent', () => {
+  it('should render empty content when given title and message', () => {
+    render(<EmptyContent title='Fake title' message='there is empty content' />);
+    expect(screen.getByText(/fake title/i)).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (K-Chow) [reply frontend/__tests__/components/Common/EmptyContent/EmptyContent.test.tsx:7]: fixed

File: frontend/src/hooks/useVerifyBoardEffect.ts (hunk 30 above)
Comment (K-Chow) [reply frontend/src/hooks/useVerifyBoardEffect.ts]: do not proposal for use， error is in a different hierarchy

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 32]
```diff
@@ -110,124 +125,109 @@ describe('Board', () => {
     expect(screen.getByText(EMAIL_REQUIRE_ERROR_MESSAGE)).toBeVisible();
   });
 
-  it('should clear other fields information when change board field selection', () => {
-    setup();
+  it.skip('should clear other fields information when change board field selection', async () => {
```
[the remark this answers is not in the document]
Comment (K-Chow) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx]: OK. removed

File: frontend/__tests__/client/BoardClient.test.ts [hunk 33]
```diff
@@ -27,7 +23,7 @@ describe('verify board request', () => {
   });
 
   it('should isBoardVerify is true when select classic jira and board verify response status 200', async () => {
```
[the remark this answers is not in the document]
Comment (K-Chow) [reply frontend/__tests__/client/BoardClient.test.ts]: fixed

File: frontend/src/context/config/configSlice.ts [hunk 34]
```diff
@@ -96,7 +96,7 @@ export const configSlice = createSlice({
             ? null
             : MESSAGE.CONFIG_PAGE_VERIFY_IMPORT_ERROR;
       }
-      state.board.config = action.payload.board || state.board.config;
+      state.board.config = merge(action.payload.board, { type: 'jira' });
```
Comment (SimonTal) [inline frontend/src/context/config/configSlice.ts:99]: extract 'jira' to enum

**Pull request merged**